addons/survey/models/survey.py

```diff
@@ -1037,9 +983,10 @@ def save_lines(self, cr, uid, user_input_id, question, post, answer_tag,
             _logger.error(question.type + ": This type of question has no saving function")
             return False
         else:
-            saver(cr, uid, user_input_id, question, post, answer_tag, context=context)
+            saver(user_input_id, question, post, answer_tag)
 
-    def save_line_free_text(self, cr, uid, user_input_id, question, post, answer_tag, context=None):
+    @api.model
+    def save_line_free_text(self, user_input_id, question, post, answer_tag):
         vals = {
             'user_input_id': user_input_id,
             'question_id': question.id,
```

Comment:
@jas-odoo remove write to page_id and survey_id in this method and all the following ones (they are related fields)